packages/app-mobile/components/get-responsive-value.test.ts

```diff
@@ -0,0 +1,265 @@
+	});
+
+	test('Should return correct value when only sm value is specified (Object Syntax)', () => {
+		const correcOutput = 40;
+		const input = { sm: 40 };
+		let result: number;
+
+		for (let i = 0; i < 13; i++) {
+			result = getResponsiveValue(input);
+			expect(result).toBe(correcOutput);
+		}
+	});
+
+	test('Should return correct value when only sm value is specified (Array Syntax)', () => {
+		const correcOutput = 40;
+		const input = [40];
+		let result: number;
+
+		for (let i = 0; i < 13; i++) {
+			result = getResponsiveValue(input);
+			expect(result).toBe(correcOutput);
+		}
+	});
+
+	test('Should return correct value when sm and md values are specified (Object Syntax)', () => {
+		const input = { sm: 40, md: 70 };
+		const correctOutputs = [40, 40, 70, 70, 70, 70, 70, 70, 70, 70, 70, 70, 70];
+		let result: number;
+
+		for (let i = 0; i < 13; i++) {
+			result = getResponsiveValue(input);
+			expect(result).toBe(correctOutputs[i]);
+		}
+	});
+
+	test('Should return correct value when sm and md values are specified (Array Syntax)', () => {
+		const input = [40, 70];
+		const correctOutputs = [40, 40, 70, 70, 70, 70, 70, 70, 70, 70, 70, 70, 70];
+		let result: number;
+
+		for (let i = 0; i < 13; i++) {
+			result = getResponsiveValue(input);
+			expect(result).toBe(correctOutputs[i]);
+		}
+	});
+
+	test('Should return correct value when sm, md and lg values are specified (Object Syntax)', () => {
+		const input = { sm: 40, md: 70, lg: 90 };
+		const correctOutputs = [40, 40, 70, 70, 70, 90, 90, 90, 90, 90, 90, 90, 90];
+		let result: number;
+
+		for (let i = 0; i < 13; i++) {
+			result = getResponsiveValue(input);
+			expect(result).toBe(correctOutputs[i]);
+		}
+	});
+
+	test('Should return correct value when sm, md and lg values are specified (Array Syntax)', () => {
+		const input = [40, 70, 90];
+		const correctOutputs = [40, 40, 70, 70, 70, 90, 90, 90, 90, 90, 90, 90, 90];
+		let result: number;
+
+		for (let i = 0; i < 13; i++) {
+			result = getResponsiveValue(input);
+			expect(result).toBe(correctOutputs[i]);
+		}
+	});
+
+	test('Should return correct value when sm, md, lg and xl values are specified (Object Syntax)', () => {
+		const input = { sm: 40, md: 70, lg: 90, xl: 110 };
+		const correctOutputs = [40, 40, 70, 70, 70, 90, 90, 90, 110, 110, 110, 110, 110];
+		let result: number;
+
+		for (let i = 0; i < 13; i++) {
+			result = getResponsiveValue(input);
+			expect(result).toBe(correctOutputs[i]);
+		}
+	});
+
+	test('Should return correct value when sm, md, lg and xl values are specified (Array Syntax)', () => {
+		const input = [40, 70, 90, 110];
+		const correctOutputs = [40, 40, 70, 70, 70, 90, 90, 90, 110, 110, 110, 110, 110];
+		let result: number;
+
+		for (let i = 0; i < 13; i++) {
+			result = getResponsiveValue(input);
+			expect(result).toBe(correctOutputs[i]);
+		}
+	});
+
+	test('Should return correct value when sm, md, lg, xl and xxl values are specified (Object Syntax)', () => {
```

Comment:
Why do we support both Object and Array syntax?

Reply:
I figured it's more convenient to use an array. But the object syntax allows omitting certain breakpoints, and that can't be done using an array. Also, it might have been influenced by something I've seen in a React UI library (Chakra UI).

Reply:
Less is more especially when it comes to internal APIs, so I would prefer if it supports only one type of input. Perhaps the Object one? Since it makes the code more self-documenting.